### PR TITLE
bitcoin: 25.1 -> 26.0

### PR DESCRIPTION
**Pull request by prusnak**
## Description of changes

- new major release
- release notes: https://bitcoincore.org/en/releases/26.0/ + https://github.com/bitcoin/bitcoin/blob/26.x/doc/release-notes.md

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by prusnak** [comment]
Darwin builds are failing, but these are not regressions - I just checked and 25.1 fail on darwin too ... :-/

**Review by RaghavSood — APPROVED** [review]
Build failures(darwin) look related to bitcoin/bitcoin/pull/27033 and bitcoin/bitcoin/issues/26869, I might take a stab at it over the weekend

**Comment by 0xB10C** [comment]
I remember running into

```
  CXX      libbitcoin_node_a-blockencodings.o
In file included from blockencodings.cpp:15:
In file included from ./txmempool.h:9:
In file included from ./coins.h:10:
In file included from ./core_memusage.h:10:
In file included from ./memusage.h:10:
./support/allocators/pool.h:161:27: error: aligned allocation function of type 'void *(std::size_t, std::align_val_t)' is only available on macOS 10.13 or newer
        void* storage = ::operator new (m_chunk_size_bytes, std::align_val_t{ELEM_ALIGN_BYTES});
                          ^
./support/allocators/pool.h:161:27: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
./support/allocators/pool.h:204:15: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.13 or newer
            ::operator delete ((void*)chunk, std::align_val_t{ELEM_ALIGN_BYTES});
              ^
./support/allocators/pool.h:204:15: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
./support/allocators/pool.h:235:18: error: aligned allocation function of type 'void *(std::size_t, std::align_val_t)' is only available on macOS 10.13 or newer
        return ::operator new (bytes, std::align_val_t{alignment});
                 ^
./support/allocators/pool.h:235:18: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
./support/allocators/pool.h:250:15: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.13 or newer
            ::operator delete (p, std::align_val_t{alignment});
              ^
./support/allocators/pool.h:250:15: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
4 errors generated.
```

on a local x86_64-darwin machine the other day, too. I think it's an issue with NixOS clang - though I haven't looked into it any deeper.

**Review by 0xB10C — APPROVED** [review]

File: pkgs/applications/blockchains/bitcoin/default.nix [hunk 1]
```diff
@@ -33,14 +33,14 @@ let
...
-  version = "25.1";
+  version = "26.0";
 
   src = fetchurl {
     urls = [
       "https://bitcoincore.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
     ];
     # hash retrieved from signed SHA256SUMS
-    sha256 = "bec2a598d8dfa8c2365b77f13012a733ec84b8c30386343b7ac1996e901198c9";
+    sha256 = "ab1d99276e28db62d1d9f3901e85ac358d7f1ebcb942d348a9c4e46f0fcdc0a1";
```
Comment (0xB10C) [inline pkgs/applications/blockchains/bitcoin/default.nix:43]: matches

**Review by 0xB10C — APPROVED** [review]

**Comment by prusnak** [comment]
> Hope to fix darwin builds with #272716. Might make sense to get that in first and backported?

There still another issue with tests failing on darwin. Hopefully it will manifest in the CI run for https://github.com/NixOS/nixpkgs/pull/272716 - we just have to wait.

**Comment by prusnak** [comment]
I am going to try whether the unmerged patch from https://github.com/bitcoin/bitcoin/pull/26915 fixes our issue.

**Comment by prusnak** [comment]
> No, it doesn't. The issue is in `wallet/test/wallet_tests.cpp.test` not in Python functional tests.

Opened the upstream issue here: https://github.com/bitcoin/bitcoin/issues/29026

**Comment by fanquake** [comment]
> Hope to fix darwin builds with https://github.com/NixOS/nixpkgs/pull/272716. Might make sense to get that in first and backported?

Note that the deployment target for 26.0 is now also macOS 11.0+.

**Comment by prusnak** [comment]
Going to merge and backport this, since bitcoin is broken on darwin already.

**Pull request merged**